semtypes/src/main/java/io/ballerina/types/typeops/ListProj.java

```diff
@@ -113,32 +119,30 @@ static SemType listProjPath(Context cx, SubtypeData k, Conjunction pos, Conjunct
                     Atom d = p.atom;
                     p = p.next;
                     lt = cx.listAtomType(d);
-                    TwoTuple intersected = listIntersectWith(members, rest,
-                            lt.members, lt.rest);
+                    TwoTuple intersected = listIntersectWith(cx.env, members, rest, lt.members(), lt.rest());
                     if (intersected == null) {
                         return NEVER;
                     }
                     members = (FixedLengthArray) intersected.item1;
-                    rest = (SemType) intersected.item2;
+                    rest = (CellSemType) intersected.item2;
                 }
             }
             if (fixedArrayAnyEmpty(cx, members)) {
                 return NEVER;
             }
             // Ensure that we can use isNever on rest in listInhabited
-            if (rest != NEVER && isEmpty(cx, rest)) {
-                rest = NEVER;
+            if (!Core.isNever(cellInnerVal(rest)) && isEmpty(cx, rest)) {
+                rest = CellSubtype.roCellContaining(cx.env, NEVER);
```

Comment:
Since the type of the cell is `NEVER` my understanding is it doesn't make a difference,  in the sense it is effectively immutable because you can't mutate something whose type is `NEVER`. Decided to do this so doing `cellInhabited` would be cleaner (since when creating list if your rest is `NEVER` we make that cell immutable). However even if we leave this as `mut = LIMITED` it should still do the right thing, (also not entirely sure if this ever reaches `cellInhabited` anyway based on the comment above the code). Do you think it is worth it to change this or revert back to what we have in nBallerina?

Reply:
+1 to keep this way.